vortex-layout/src/layouts/zoned/writer.rs

```diff
@@ -115,29 +109,30 @@ where
         )
         .sendable();
 
-        let ctx = ctx.clone();
         let block_size = self.options.block_size;
+
+        // We create a new SequencePointer for the stats table so that we can write it just
+        // before the end of the file.
+        let (stats_eof, eof) = eof.split();
+
         let data_layout = self
             .child
-            .write_stream(&ctx, sequence_writer.clone(), stream)
+            .write_stream(ctx, segment_sink, stream, eof, handle.clone())
```

Comment:
This looks a bit funny to me. I reckon this would only work if you don't use the `eof` at all anywhere in child, and its children. Because `stats_eof` is still alive, calling collapse on any sequence id derived from eof would block forever